### PR TITLE
TC03 - User cannot log into Railway with invalid password 

### PR DESCRIPTION
**Pull request by cvan8397**

### CODE REVIEW

File: src/test/resources/testNG.xml [hunk 1]
```diff
@@ -3,7 +3,8 @@
     <test name="Chrome Test">
         <parameter name="browser" value="chrome"/>
         <classes>
-            <class name="tests/TC01.java" />
+            <class name="tests.TC03" />
+            <class name="tests.TC01" />
         </classes>
     </test>
 </suite>
```
Comment (tanmle) [inline src/test/resources/testNG.xml]: break line.

File: src/main/java/page_objects/GeneralPage.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import org.openqa.selenium.By;
+
+public class GeneralPage {
+    private final Tab loginTab = new Tab(By.cssSelector("a[href='/Account/Login.cshtml']"));
+    private final Tab registerTab = new Tab(By.cssSelector("a[href='/Account/Register.cshtml']"));
+    private final Tab bookTicketTab = new Tab(By.cssSelector("a[href='/Page/BookTicketPage.cshtml']>span"));
+    private final Tab logOutTab = new Tab(By.cssSelector("a[href='/Account/Logout']>span"));
+    private final Tab changePwdTab = new Tab(By.cssSelector("a[href='/Account/ChangePassword.cshtml']"));
+    private final Tab MyTicketTab = new Tab(By.cssSelector("a[href='/Page/ManageTicket.cshtml']"));
+    private final Label lblWelcomeMessage = new Label(By.cssSelector(".account"));
```
Comment (tanmle) [inline src/main/java/page_objects/GeneralPage.java]: how about By.className?

File: src/main/java/page_objects/GeneralPage.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package page_objects;
+
+import elements.Label;
+import elements.Tab;
+import org.openqa.selenium.By;
+
+public class GeneralPage {
+    private final Tab loginTab = new Tab(By.cssSelector("a[href='/Account/Login.cshtml']"));
+    private final Tab registerTab = new Tab(By.cssSelector("a[href='/Account/Register.cshtml']"));
+    private final Tab bookTicketTab = new Tab(By.cssSelector("a[href='/Page/BookTicketPage.cshtml']>span"));
```
Comment (tanmle) [inline src/main/java/page_objects/GeneralPage.java]: why do we need `span` for this while the other doesn't have?

File: src/test/java/tests/TestBase.java [hunk 4]
```diff
@@ -1,14 +1,24 @@
...
-    public void setup() {
+    @Parameters("browser")
+    public void setup(@Optional("chrome") String browser) {
+        Log.info("Pre-condition");
+        BrowserHelper.startBrowser(BrowserHelper.DriverType.valueOf(browser.toUpperCase()));
+        BrowserHelper.navigateToUrl(Constant.RAILWAY_URL);
     }
 
     @AfterMethod
     public void closePage() {
```
Comment (tanmle) [inline src/test/java/tests/TestBase.java]: I already commented about this, this is not `closePage`

**Review by tanmle — CHANGES_REQUESTED** [review]

File: src/main/java/helper/Constant.java [hunk 5]
```diff
@@ -1,6 +1,9 @@
 package helper;
 
 public class Constant {
-
-
+    public static final String RAILWAY_URL = "http://www.railway2.somee.com/Page/HomePage.cshtml";
+    public static final String USERNAME = "caivan.83@gmail.com";
+    public static final String PASSWORD = "123123123";
+    public static final String BLANK = "";
```
Comment (tanmle) [inline src/main/java/helper/Constant.java]: why do we need a constant for just a BLANK?

File: src/main/java/models/Account.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
+package models;
+
+public class Account {
+    private String username;
+    private String password;
+    private String confirmPassword;
```
Comment (tanmle) [inline src/main/java/models/Account.java]: this is not a property for account

**Review by tanmle — CHANGES_REQUESTED** [review]

File: src/main/java/models/Account.java (hunk 6 above)
Comment (fey031) [reply src/main/java/models/Account.java]: @tanmle I think it should be registerAccount

**Comment by cvan8397** [comment]
@tanmle I have fixed my code based on your request. Please review and let me know if any problems

**Review by tanmle — APPROVED** [review]

**Review by LoliSlayerMC — APPROVED** [review]
LGTM

**Pull request merged**